### PR TITLE
dotnet-test: add unit-under-test + behaviors quality cue to code-testing-generator

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
## What

Adds a quality cue to the orchestrator agent (`code-testing-generator`) and small companion changes in `code-testing-implementer` and `code-testing-fixer` that ask each subagent to extract the **unit-under-test contract** and the **behaviors to test** before writing or fixing tests.

The cue is **orchestrator-level** (a verification gate after dispatch) with **bounded re-dispatch**, not new content rules baked into planner/researcher/implementer prompts. Same structural pattern as the existing research-cue.

## Why

CTA was losing rubric quality (`nice%`, `agg%`) to the vanilla CLI baseline despite matching coverage. The proximate cause (verified via per-instance patch inspection) was that the planner under-specified what the test was supposed to demonstrate, and the implementer wrote weak assertions (e.g. `assert.NotNil(err)` instead of `assert.Contains(err.Error(), "must not be empty")`).

This cue makes the contract explicit before code is written, without prescribing assertion style — so it's framework-agnostic and not benchmark-specific.

### CODE REVIEW

File: plugins/dotnet-test/agents/code-testing-generator.agent.md [hunk 1]
```diff
@@ -28,13 +114,32 @@ Understand what the user wants: scope (project, files, classes), priority areas,
...
 
+### Step 1b: Mandatory initial researcher dispatch (every strategy, no exceptions)
+
+Before any other CTA dispatch, dispatch the researcher once to populate `.testagent/research.md`:
+
+```text
+task({
+  agent_type: "dotnet-test:code-testing-researcher",
+  name: "researcher",
+  prompt: "Initial scoping research for test generation. Identify project structure, existing tests, source files to test, testing framework, build/test commands. Write findings to .testagent/research.md."
```
[the remark this answers is not in the document]
Comment (YuliiaKovalova) [reply plugins/dotnet-test/agents/code-testing-generator.agent.md]: Fixed in c93a62. Updated the Step 1b prompt to explicitly require (1) the unit-under-test contract with a ile:line citation and (2) behaviors/edge cases — so the verification gate at line 129-134 should rarely need to trigger a re-dispatch. Also addressed the duplicate at Step 3 (was line 161): renamed to **Deep Research Phase**, scoped to Single pass / Iterative only (skipped for Direct), and changed the prompt from overwriting 
esearch.md to extending it — Step 1b's findings are preserved.

File: plugins/dotnet-test/agents/code-testing-generator.agent.md [hunk 2]
```diff
@@ -120,10 +252,31 @@ After the previous phases complete, check for uncovered source files:
...
+
+Then re-run planner (writing `.testagent/plan-2.md`) and implementer for the gap phase, followed by builder/tester/fixer cycles. Do this at most once per run; if the second iteration also leaves gaps, list them in the final report rather than looping further.
+
+### Step 9: Validate Diff and Clean Up
+
+Before reporting success, verify the patch contains only legitimate test changes and remove pipeline scratch state. Always dispatch the builder as a validator (Rule 5 — never run cleanup commands inline via `terminal`). This applies to ALL strategies including Direct:
+
+```text
+task({
+  agent_type: "dotnet-test:code-testing-builder",
```
[the remark this answers is not in the document]
Comment (YuliiaKovalova) [reply plugins/dotnet-test/agents/code-testing-generator.agent.md]: Fixed in c93a62. Stopped overloading code-testing-builder as a validator. Step 9 now describes the diff/cleanup steps as **orchestrator-direct work** (git status --porcelain, git diff --name-only HEAD, removing .testagent/, reverting modified non-test files). Added a note clarifying that Rule 5 forbids inline 	erminal for build/test only — git/filesystem hygiene is fair game for the orchestrator.

File: plugins/dotnet-test/agents/code-testing-generator.agent.md [hunk 3]
```diff
@@ -28,13 +114,32 @@ Understand what the user wants: scope (project, files, classes), priority areas,
...
 | **Single pass** | A moderate scope (couple projects or modules) that a single Research → Plan → Implement cycle can cover | Execute Steps 3-8 once, then proceed to Step 9. |
 | **Iterative** | A large scope or ambitious coverage target that one pass cannot satisfy | Execute Steps 3-8, then re-evaluate coverage. If the target is not met, repeat Steps 3-8 with a narrowed focus on remaining gaps. Use unique names for each iteration's `.testagent/` documents (e.g., `research-2.md`, `plan-2.md`) so earlier results are not overwritten. Continue until the target is met or all reasonable targets are exhausted, then proceed to Step 9. |
```
[the remark this answers is not in the document]
Comment (YuliiaKovalova) [reply plugins/dotnet-test/agents/code-testing-generator.agent.md]: Fixed in c93a62. Updated all step cross-references: `Steps 6-9` → `Steps 6-10` in the Direct strategy cell and the All-strategies-MUST line; `proceed to Step 9` → `proceed to Steps 9-10` in Single pass / Iterative cells. The summary description also now reads `...coverage gap iteration, diff validation/cleanup, and reporting` to match Step 9 + Step 10.

File: plugins/dotnet-test/agents/code-testing-generator.agent.md [hunk 4]
```diff
@@ -55,63 +160,90 @@ Based on the request scope, pick exactly one strategy and follow it:
...
-- **Go**: `go build ./...` from module root
-- **Rust**: `cargo build`
+Always dispatch the builder (Rule 5 — never run the build inline via `terminal`). This applies to ALL strategies including Direct:
 
-If it fails, call the `code-testing-fixer`, rebuild, retry up to 3 times.
+```text
+task({
+  agent_type: "dotnet-test:code-testing-builder",
+  name: "builder",
+  prompt: "Run a full, non-incremental workspace build. .NET: 'dotnet build *.sln --no-incremental' with NO --framework flag (must build all target frameworks). TypeScript: 'npx tsc --noEmit' from workspace root. Go: 'go build ./...' from module root. Rust: 'cargo build'. Report any errors."
```
[the remark this answers is not in the document]
Comment (YuliiaKovalova) [reply plugins/dotnet-test/agents/code-testing-generator.agent.md]: Fixed in c93a62. Replaced `'dotnet build *.sln --no-incremental'` with `'dotnet build --no-incremental' from the repo root` (auto-discovers a single .sln) plus a fallback note `If the repo contains a .sln/.slnx, use 'dotnet build <solution>.sln --no-incremental'`. This now matches skills/code-testing-extensions/extensions/dotnet.md.

File: plugins/dotnet-test/agents/code-testing-fixer.agent.md [hunk 5]
```diff
@@ -1,9 +1,10 @@
 ---
 description: >-
-  Fixes compilation errors in source or test files.
+  Fixes compilation errors and failing tests in source or test files.
 
   Use when: resolving build errors, fixing CS/TS error codes, adding missing
-  imports, correcting type mismatches, fixing compilation failures.
+  imports, correcting type mismatches, fixing compilation failures, OR
```
[the remark this answers is not in the document]
Comment (YuliiaKovalova) [reply plugins/dotnet-test/agents/code-testing-fixer.agent.md:6]: Fixed in c93a62. Updated the fixer body to match the front-matter description: opening line now reads `You fix compilation errors **and failing tests** in code files`, and the Mission section explicitly lists the two failure modes — compilation errors **and** failing test assertions (with explicit no-Ignore/no-Skip/no-production-rewrite guardrails). Step 1 (Parse Error Information) also broadened to cover both modes.

**Review by Evangelink — APPROVED** [review]
